cluster-autoscaler/core/scale_down.go

```diff
@@ -1065,7 +1065,34 @@ func (sd *ScaleDown) scheduleDeleteEmptyNodes(emptyNodes []*apiv1.Node, client k
 			return deletedNodes, errors.ToAutoscalerError(errors.ApiCallError, taintErr)
 		}
 		deletedNodes = append(deletedNodes, node)
-		go func(nodeToDelete *apiv1.Node, nodeGroupForDeletedNode cloudprovider.NodeGroup) {
+		go func(nodeToDelete *apiv1.Node, nodeGroupForDeletedNode cloudprovider.NodeGroup, daemonSetEvictionForEmptyNodes bool) {
+			if daemonSetEvictionForEmptyNodes {
+				nodeInfo, _ := sd.context.ClusterSnapshot.NodeInfos().Get(nodeToDelete.Name)
```

Comment:
WDYT about extracting this logic to a separate function/method? This would allow us to easily test it, leaving only minimal changes here - 1 line if.

Reply:
Do you suggest to add a new method to scale down object?

Reply:
+1